server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeFileTask.java

```diff
@@ -208,15 +208,27 @@ private void moveMergedToOld(TsFileResource seqFile) throws IOException {
     }
   }
 
-  private void updateStartTimeAndEndTime(TsFileResource seqFile, TsFileIOWriter fileWriter) {
+  private void updateStartTimeAndEndTime(TsFileResource seqFile, TsFileSequenceReader fileReader)
+      throws IOException {
     // TODO change to get one timeseries block each time
-    for (Entry<String, List<ChunkMetadata>> deviceChunkMetadataEntry :
-        fileWriter.getDeviceChunkMetadataMap().entrySet()) {
-      String device = deviceChunkMetadataEntry.getKey();
-      for (ChunkMetadata chunkMetadata : deviceChunkMetadataEntry.getValue()) {
-        seqFile.updateStartTime(device, chunkMetadata.getStartTime());
-        seqFile.updateEndTime(device, chunkMetadata.getEndTime());
+    List<String> devices = fileReader.getAllDevices();
+    for (String device : devices) {
+      Map<String, List<ChunkMetadata>> chunkMetadataListMap =
+          fileReader.readChunkMetadataInDevice(device);
+      long minStartTime = Long.MAX_VALUE;
+      long maxEndTime = Long.MIN_VALUE;
+      for (List<ChunkMetadata> chunkMetadataList : chunkMetadataListMap.values()) {
+        for (ChunkMetadata chunkMetadata : chunkMetadataList) {
+          if (chunkMetadata.getStartTime() < minStartTime) {
+            minStartTime = chunkMetadata.getStartTime();
+          }
+          if (chunkMetadata.getEndTime() > maxEndTime) {
+            maxEndTime = chunkMetadata.getEndTime();
+          }
+        }
       }
+      seqFile.putStartTime(device, minStartTime);
+      seqFile.putEndTime(device, maxEndTime);
```

Comment:
I do not think the range of the new file can be directly put into the resource.
Consider a compaction with a seq file containing three chunks, [0, 7] [10, 15] [20, 29], and an unseq file with one chunk, [17,28]. The new file (merged file) will contain only a chunk, [17, 29] if a full merge is not required, and the chunk will be appended to the old seq file as merged chunks are the minority. If you just use the time range from the new file, which is [17, 29], to put into the new TsFileResource,  the result will not be accurate.

Reply:
okay